core/deco.c

```diff
@@ -293,7 +292,7 @@ double tissue_tolerance_calc(struct deco_state *ds, const struct dive *dive, dou
 /*
  * Return Buehlmann factor for a particular period and tissue index.
  */
-static double factor(int period_in_seconds, int ci, enum inertgas gas)
+static double factor(int period_in_seconds, int ci, enum gas_component gas)
 {
 	if (period_in_seconds == 1) {
 		if (gas == N2)
```

Comment:
Now, that there are more than two possible values, maybe this should be a switch statement (with the O2 case either throwing an error or giving a result of 1.0 corresponding to an infinite half-time for oxygen).

Reply:
Fair point. However, since this is a `static` function, it is not accessible outside the source file, so it should be easy to check that no invalid values are provided.